### PR TITLE
Add best practices document

### PR DESCRIPTION
**Pull request by orkunkl**

### CODE REVIEW

File: BESTPRACTICES.md [hunk 1]
```diff
@@ -0,0 +1,18 @@
+# Best Practices
+---
+## Weave 
+> Is weave.Metadata metadata always required?
+
+This allows you to upgrade the code on chain without downtime
```
Comment (ethanfrey) [inline BESTPRACTICES.md]: Add link to `weave/migrations`

File: BESTPRACTICES.md [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+
+> What is the best practices for using proto value types on modeling currency amount, IDs, timestamp?  
+
+ - Currency example [weave/x/coin](https://github.com/iov-one/weave/tree/master/coin)
+ - For incremetal ID use `orm.IDGenBucket`
```
Comment (ethanfrey) [inline BESTPRACTICES.md]: We can use new `morm/ModelBucket` now :)

I'd be happy for your feedback on that

Comment (orkunkl) [reply BESTPRACTICES.md]: Read [iov-one/weave#712](https://github.com/iov-one/weave/issues/712). I am delighted to raise new concerns for you guys :relaxed: I will add a _TODO_ line until ```morm/ModelBucket``` merged into weave. Want to keep this document unified for future newcomers.

**Comment by ethanfrey** [comment]
Nice additions.

File: BESTPRACTICES.md (hunk 1 above)
Comment (ethanfrey) [reply BESTPRACTICES.md]: The docs.go file in there is actually quite insightful, probably best to link directly to it

**Pull request merged**